### PR TITLE
Add basic e2e test

### PR DESCRIPTION
**Pull request by miguelsorianod**
This PR adds basic e2e test that deploy an APIcast with an embedded configuration.
It basically checks that the operator and the APIcast deployment itself are started correctly.

In a future PR I intend to improve the test / add a new test by trying to look some additional elements (service / try to perform some request to APIcast somehow?)

### CODE REVIEW

**Comment by miguelsorianod** [comment]
Ready for review @eguzki

File: test/e2e/apicast_test.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	start = time.Now()
+
+	err = f.Client.Create(context.TODO(), apicast, &framework.CleanupOptions{TestContext: ctx, Timeout: 5 * time.Minute, RetryInterval: retryInterval})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	elapsed = time.Since(start)
+
+	err = createAPIcastEmbeddedConfigurationSecret(t, ctx, f, namespace)
```
Comment (eguzki) [inline test/e2e/apicast_test.go]: create the secret before deploying apicast CR? You do not need to implement waiting for secret to be available, apicast deployment resource will do that for you, if I am not wrong.

**Comment by eguzki** [comment]
good basic e2e test :+1:

File: .circleci/config.yml [hunk 2]
```diff
@@ -102,7 +158,9 @@ jobs:
       - run: make licenses-check
 
   tag-operator-image-master:
-    machine: true
+    machine:
+      image: circleci/classic:latest
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply .circleci/config.yml]: The reasons I've made it explicit are:

* It is marked as required in the CircleCI configuration reference:
https://circleci.com/docs/2.0/configuration-reference/#docker--machine--macos--windows-executor
* The linter that my IDE (VSCode) seems to use for CircleCI detects this as required as well and complains marking it as an error with 'missing property: "image"'

File: .circleci/config.yml [hunk 3]
```diff
@@ -70,7 +127,7 @@ commands:
       - run:
           name: Build Operator
           command: |
-              make build IMAGE=172.30.1.1:5000/openshift/apicast-operator VERSION=test
+              make build IMAGE=172.30.1.1:5000/openshift/apicast-operator VERSION=${CIRCLE_SHA1}
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply .circleci/config.yml]: Both things would work. I don't really see a problem.

File: .circleci/config.yml [hunk 4]
```diff
@@ -79,18 +136,17 @@ jobs:
...
           name: Sharing requirements to downstream job
           command: |
             mkdir -p /tmp/workspace/images
-            docker save -o /tmp/workspace/images/operator-image.tar 172.30.1.1:5000/openshift/apicast-operator:test
+            docker save -o /tmp/workspace/images/3scaleapicastoperator-image.tar 172.30.1.1:5000/openshift/apicast-operator:${CIRCLE_SHA1}
       - persist_to_workspace:
           root: /tmp/workspace
           paths:
-            - images/operator-image.tar
+            - images/3scaleapicastoperator-image.tar
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply .circleci/config.yml]: I'm going to change it :+1:

File: test/e2e/apicast_test.go (hunk 1 above)
Comment (miguelsorianod) [reply test/e2e/apicast_test.go]: You are correct. You don't need to create the secret before deploying APIcast as the operator would return errors and requeue the event until it finds it.

That said, I don't see a problem with precreating the secret and waiting for it so we define an order. We just make sure in this test that order events happen.

File: .circleci/config.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply .circleci/config.yml]: Changed

**Comment by miguelsorianod** [comment]
Ready for review @eguzki

File: .circleci/config.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (eguzki) [reply .circleci/config.yml]: it is required and old default. :+1:

File: test/e2e/apicast_test.go (hunk 1 above)
Comment (eguzki) [reply test/e2e/apicast_test.go]: My point is, if deployment waits, why implement it?

IIRC when deployment has env var from secret and secret does not start, deployment will wait until that secret exist (doing internal polling I assume). Just for simplicity. 

Up to you

**Pull request merged**